neo/core/basesignal.py

```diff
@@ -91,10 +98,16 @@ def _rescale(self, signal, units=None):
                 raise ValueError("Units must be specified")
         elif isinstance(signal, pq.Quantity):
             # could improve this test, what if units is a string?
-            if units != signal.units:
+            # This test should be safe now
+            if pq.quantity.validate_dimensionality(units) != signal.dimensionality:
                 signal = signal.rescale(units)
         return signal
 
+    def rescale(self, units):
+        obj = super(BaseSignal, self).rescale(units)
+        obj.channel_index = self.channel_index
```

Comment:
Segment is set in DataObject.rescale, because this needs to be done for all DataObjects anyway. ChannelIndex is more specific and not required for all DataObjects, so it needs to be treated separately.